internal/factsengine/gatherers/sysctl.go

```diff
@@ -0,0 +1,118 @@
+package gatherers
+
+import (
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	"github.com/trento-project/agent/pkg/utils"
+)
+
+const (
+	SysctlGathererName = "sysctl"
+)
+
+// nolint:gochecknoglobals
+var (
+	SysctlValueNotFound = entities.FactGatheringError{
+		Type:    "sysctl-value-not-found",
+		Message: "requested value not found in sysctl output",
+	}
+
+	SysctlCommandError = entities.FactGatheringError{
+		Type:    "sysctl-cmd-error",
+		Message: "error executing sysctl command",
+	}
+
+	SysctlMissingArgument = entities.FactGatheringError{
+		Type:    "sysctl-missing-argument",
+		Message: "missing required argument",
+	}
+)
+
+type SysctlGatherer struct {
+	executor utils.CommandExecutor
+}
+
+func NewDefaultSysctlGatherer() *SysctlGatherer {
+	return NewSysctlGatherer(utils.Executor{})
+}
+
+func NewSysctlGatherer(executor utils.CommandExecutor) *SysctlGatherer {
+	return &SysctlGatherer{
+		executor: executor,
+	}
+}
+
+func (s *SysctlGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
+	log.Infof("Starting sysctl facts gathering process")
+
+	output, err := s.executor.Exec("sysctl", "-a")
+	if err != nil {
+		return nil, SysctlCommandError.Wrap(err.Error())
+	}
+
+	sysctlMap := sysctlOutputToMap(output)
+	for _, factReq := range factsRequests {
+		var fact entities.Fact
+
+		if len(factReq.Argument) == 0 {
+			log.Error(SysctlMissingArgument.Message)
+			fact = entities.NewFactGatheredWithError(factReq, &SysctlMissingArgument)
+		} else if value, err := sysctlMap.GetValue(factReq.Argument); err == nil {
+			fact = entities.NewFactGatheredWithRequest(factReq, value)
+		} else {
+			gatheringError := SysctlValueNotFound.Wrap(factReq.Argument)
+			log.Error(gatheringError)
+			fact = entities.NewFactGatheredWithError(factReq, gatheringError)
+		}
+
+		facts = append(facts, fact)
+	}
+
+	log.Infof("Requested %s facts gathered", CorosyncCmapCtlGathererName)
+	return facts, nil
+}
+
+func sysctlOutputToMap(output []byte) *entities.FactValueMap {
```

Comment:
Could we try to use `envparse.Parse(output)` maybe?
At the end the output looks like an `.env` file format.
I know that this wouldn't create a map of maps, but I don't think we would need such a thing in this case. I expect the user putting the whole needed line to get individual values

I would try to change, you have the tests there to confirm that everything went right

PD: I know that this is based in the `corosynccmapctl` gatherer, but i think we could make it simpler this way

Reply:
Indeed it's based on the corosync one, I saw a comment in confluence mentioning that we might want to expose `sysctl -a` (with no arguments) as a map, so I used the same approach. In the end apparently we don't need this but if we do in the future, the current approach would be more powerful. In any case, no hard opinions on this.

Reply:
Anyway, do we need all of this `switch` code with the `SplitN`?
I think the `corosynccmapctl` code is cleaner, and does exactly the same (99% the same)

```
parts := strings.Split(line, "=")
key := parts[0]
value := parts[1]
```